plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -102,7 +102,7 @@ protected Control createDialogArea(Composite parent) {
           @Override
           public int getMessageType(ValidationStatusProvider statusProvider) {
             int type = super.getMessageType(statusProvider);
-            setValid(type != IMessageProvider.ERROR);
+            setValid(type != IMessageProvider.ERROR && apiFactory.isLoggedIn());
```

Comment:
`isLoggedIn()` is removed from the `IGoogleApiFactory` below, but then used here. I'm confused - how does this compile? Am I mistaken on which interface is being used here?

Reply:
Oh, it seems like this slipped in when merging the `gcloud-creds-cli` branch. I changed it to `hasCredentialsSet()`.